test/e2e/nftables_test.go

```diff
@@ -45,20 +62,36 @@ var _ = Describe("Nftables", func() {
 			masterNFT, err = AddPortsToNFTables(masterNFT, extraNFTablesFile)
 			Expect(err).NotTo(HaveOccurred())
 		}
-
 		g := new(errgroup.Group)
-		for _, node := range nodeList.Items {
-			nodeName := node.Name
-			nodeRole, err := types.GetNodeRole(&node)
-			Expect(err).ToNot(HaveOccurred())
+		clusterVersion, err := cluster.GetClusterVersion(cs)
+		Expect(err).ToNot(HaveOccurred())
+
+		currentVersion, err := semver.NewVersion(clusterVersion)
+		Expect(err).NotTo(HaveOccurred())
+
+		thresholdVersion := semver.MustParse("4.16")
+		if currentVersion.GreaterThan(thresholdVersion) { // If the version is greater than 4.16, the cluster's MachineConfiguration must be modified.
+			By("Version Greater Than 4.16 Need To Update Machine Configuration")
+			if err = updateMachineConfiguration(cs); err != nil {
+				Expect(err).ToNot(HaveOccurred())
+			}
+		}
+
+		for _, role := range nodeRoles {
```

Comment:
What if a node has two roles, 'master' and 'worker'?
The MachineConfig should not be created twice.

Reply:
If you create a MachineConfig (MC) per role, multiple MCs will be touching the same set of files (e.g. `/etc/sysconfig/nftables.conf` and nftables.service systemd unit file). Especially for `/etc/sysconfig/nftables.conf`, that means ending up with the nftables rules for a single role which is not the desired outcome.

Reply:
I changed that to be by role
Like to get the node role first and that will not have duplicate

Reply:
if it worker + master we will care to the master since its the same and it include all
on that we will hide the duplicates of creating mc on same node